src/hammer-vlsi/hammer_vlsi/hammer_vlsi_impl.py

```diff
@@ -350,6 +354,363 @@ def output_gds(self, value: str) -> None:
+        except AttributeError:
+            raise ValueError("Nothing set for the collection of schematic files (e.g. *.v or *.sp) yet")
+
+    @schematic_files.setter
+    def schematic_files(self, value: List[str]) -> None:
+        """Set the collection of schematic files (e.g. *.v or *.sp)."""
+        if not (isinstance(value, List)):
+            raise TypeError("schematic_files must be a List[str]")
+        self.attr_setter("_schematic_files", value)
+
+    @property
+    def power_nets(self) -> List[str]:
+        """
+        Get a list of power nets used in the design
+
+        :return: The list of power nets used in the design
+        """
+        try:
+            return self.attr_getter("_power_nets", None)
+        except AttributeError:
+            raise ValueError("Nothing set for the list of power nets used in the design yet")
+
+    @power_nets.setter
+    def power_nets(self, value: List[str]) -> None:
+        """ Set the input power nets used in the design"""
+        if not (isinstance(value, List)):
+            raise TypeError("power_nets must be a List[str]")
+        self.attr_setter("_power_nets", value)
+
+    @property
+    def ground_nets(self) -> List[str]:
+        """
+        Get a list of ground nets used in the design
+
+        :return: The list of ground nets used in the design
+        """
+        try:
+            return self.attr_getter("_ground_nets", None)
+        except AttributeError:
+            raise ValueError("Nothing set for the list of ground nets used in the design yet")
+
+    @ground_nets.setter
+    def ground_nets(self, value: List[str]) -> None:
+        """ Set the input ground nets used in the design"""
+        if not (isinstance(value, List)):
+            raise TypeError("ground_nets must be a List[str]")
+        self.attr_setter("_ground_nets", value)
+
+    @property
+    def hcells_list(self) -> List[str]:
+        """
+        Get the list of hierarchical cells for LVS.
+
+        :return: The list of hierarchical cells for LVS.
+        """
+        try:
+            return self.attr_getter("_hcells_list", None)
+        except AttributeError:
+            raise ValueError("Nothing set for the list of hierarchical cells for LVS yet")
+
+    @hcells_list.setter
+    def hcells_list(self, value: List[str]) -> None:
+        """Set the list of hierarchical cells for LVS."""
+        if not (isinstance(value, list)):
+            raise TypeError("hcells_list must be a List[str]")
+        self.attr_setter("_hcells_list", value)
+
+
+    ### Outputs ###
+
+    @abstractmethod
+    def erc_results_pre_waived(self) -> Dict[str, int]:
+        """ Return a Dict mapping the ERC check name to an error count (pre-waivers). """
+        pass
+
+    def signoff_results(self) -> int:
+        """ Return the count of unwaived ERC errors and LVS errors. """
+        return reduce(lambda a, b: a + b, self.erc_results().values()) + len(self.lvs_results())
```

Comment:
`reduce(lambda a, b: a + b, self.erc_results().values())` ==> `sum(self.erc_results().values())`?

Reply:
Yes, that is cleaner. Will fix.